### PR TITLE
feat: Adding Data Dashboard link to Slow Zone posts

### LINKED ISSUE #15: Add link to Data Dashboard in Slow Zone tweet
For when zones appear or are fixed, I think it'd be cool if we had a truncated URL embedded on our tweets so that folks can go straight to the Dash to see additional information about the new/former slow zone (will be even cooler when v4 is released)

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Fixes #15 

@skaplan-dev I was able to kinda test this by putting the function somewhere unconventional and doing a dry run of the bot, but curious if you have a way to test this stuff so I could visualize how it looks?

@devinmatte not sure if there's anyway to future-proof this for V4, but if so feel free to advise :+1:

### CODE REVIEW

**Comment by skaplan-dev** [comment]
> Fixes #15
> 
> @skaplan-dev I was able to kinda test this by putting the function somewhere unconventional and doing a dry run of the bot, but curious if you have a way to test this stuff so I could visualize how it looks?
> 
> @devinmatte not sure if there's anyway to future-proof this for V4, but if so feel free to advise 👍

no super great way to test this besides just printing things out locally. it might be worth us building a `test` Twitter bot for stuff like this.

**Comment by nathan-weinberg** [comment]
> > Fixes #15
> > @skaplan-dev I was able to kinda test this by putting the function somewhere unconventional and doing a dry run of the bot, but curious if you have a way to test this stuff so I could visualize how it looks?
> > @devinmatte not sure if there's anyway to future-proof this for V4, but if so feel free to advise +1
> 
> no super great way to test this besides just printing things out locally. it might be worth us building a `test` Twitter bot for stuff like this. I'd actually prefer if we tweeted the dashboard link as a response to the tweet (open to debating this). We don't have great twitter `meta` tags that add nice visualizations for dashboard links.

I'd def be open to that idea - I think visually it'd be more appealing - we might need to play with the bot logic since the reply scheme would work differently for Mastodon/Slack.

Another thing I was looking into was URL shorteners  for the Dashboard links so they might look a bit better as part of the main tweet - we would probably need to sign up for a service like [Bitly](https://bitly.com/) for that, the only native Python library I could find for that [seems abandoned](https://github.com/ellisonleao/pyshorteners/).

**Comment by devinmatte** [comment]
We should update this now with v4 links

File: utils.py [hunk 1]
```diff
@@ -105,17 +109,33 @@ def generate_post_text_map(g_sz):
...
+def generate_data_dashboard_link(sz):
+    ''' takes in a slow zone
+        returns a data dashboard link to said slow zone
+    '''
+    color = sz['color']
+    stop1 = sz['fr_id']
+    stop2 = sz['to_id']
+    start = (datetime.strptime(sz['start'], '%Y-%m-%dT%H:%M:%S') - timedelta(days=14)).strftime('%Y-%m-%d')
+    end = sz['end'].split('T')[0]
+    link = f"https://dashboard.transitmatters.org/rapidtransit?config={color},{stop1},{stop2},{start},{end}"
```
Comment (devinmatte) [inline utils.py:121]: Format for v4 is
```suggestion
    link = f"https://dashboard.transitmatters.org/{color}/trips/multi/
    ?from={stop1}&to={stop2}&startDate={start}&endDate={end}"
```

**Review by devinmatte — CHANGES_REQUESTED** [review]

**Review by devinmatte — APPROVED** [review]
LGTM. Maybe wait on @skaplan-dev to make sure we're ready to launch this

**Comment by nathan-weinberg** [comment]
I tried to test this with the test Twitter bot but was unable to do so - would someone who configured the Prod bot be able to adjust the settings to allow for write access? Can share the keys via Slack if needed

**Comment by nathan-weinberg** [comment]
Ofc we can always merge this and hope for the best :smile:

**Pull request merged**